### PR TITLE
metrics rounding & interpolation

### LINKED ISSUE #4579: Metrics fixes

### PR DESCRIPTION
**Pull request by matborowczyk**
# Description

* Short description here *

closes #4579 

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [x] Attached issue to pull request
- [x] Changelog entry
- [x] Code is clear and sufficiently documented
- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [x] Correct, in line with design
- [ ] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )

### CODE REVIEW

File: src/Slices/Dashboard/UI/Charts/LineChart.tsx [hunk 1]
```diff
@@ -47,6 +48,16 @@ export const LineChart: React.FC<LineChartProps> = ({
     }
     return value % 1 === 0 ? value : Math.round(value * 1000) / 1000;
   };
+  const chooseWhichLabelTouse = (datum) => {
```
Comment (LauraStordeur) [inline src/Slices/Dashboard/UI/Charts/LineChart.tsx:51]: `chooseWhichLabelToUse`

File: src/Slices/Dashboard/UI/helper.ts [hunk 2]
```diff
@@ -61,12 +61,79 @@ export const formatMetricsToStacked = (
...
+      }
+    } else {
+      newMetric.push(value);
+    }
+  });
+
+  return newMetric;
+};
+
+const lerp = (a, b, amount) => (1 - amount) * a + amount * b;
```
Comment (LauraStordeur) [inline src/Slices/Dashboard/UI/helper.ts:139]: maybe write out the full word, linearInterpolation :)

File: src/Slices/Dashboard/UI/helper.ts [hunk 3]
```diff
@@ -61,12 +61,79 @@ export const formatMetricsToStacked = (
...
+    if (nextNumber === -1) {
+      newMetric.push(null);
+    }
+    if (value === null && (index === 0 || index === metrics.length - 1)) {
+      newMetric.push(null);
+    } else if (value === null && newMetric[index - 1] === null) {
+      newMetric.push(null);
+    } else if (value === null && newMetric[index - 1] !== null) {
+      if (metrics[index + 1] !== null) {
+        newMetric.push(lerp(newMetric[index - 1], metrics[index + 1], 0.5));
```
Comment (LauraStordeur) [inline src/Slices/Dashboard/UI/helper.ts:120]: why 0.5 ?

File: src/Slices/Dashboard/UI/helper.ts [hunk 4]
```diff
@@ -61,12 +61,79 @@ export const formatMetricsToStacked = (
...
+ * as long as there are some boundaries) with interpolation on line between those two numbers.
+ *
+ * @param metrics
+ * @returns metrics with interpolated values instead null values
+ */
+export const interpolateMetrics = (metrics: (number | null)[]) => {
+  const newMetric: (number | null)[] = [];
+  let nextNumber = 0;
+
+  metrics.forEach((value, index) => {
```
Comment (LauraStordeur) [inline src/Slices/Dashboard/UI/helper.ts:110]: I think this piece of code can use a few comments to explain the logic and reasoning :)

File: src/Slices/Dashboard/UI/helper.ts (hunk 3 above)
Comment (matborowczyk) [reply src/Slices/Dashboard/UI/helper.ts:120]: 0.5 is used as we want the middle value(average) between these two values, but I now removed that if and moved all of the interpolation inside last else if

File: src/Slices/Dashboard/UI/helper.ts (hunk 4 above)
Comment (matborowczyk) [reply src/Slices/Dashboard/UI/helper.ts:110]: Yup, that's right, I didn't think of that :D

**Review by LauraStordeur — APPROVED** [review]

**Pull request closed without merging**